addon/ActivityStreams.js

```diff
@@ -239,7 +239,7 @@ ActivityStreams.prototype = {
     this._store.dispatch(am.actions.Response("PREFS_RESPONSE", simplePrefs.prefs));
 
     // Share
-    this._store.dispatch(am.actions.Response("SHARE_PROVIDERS_RESPONSE", this._shareProvider.socialProviders));
+    this._store.dispatch(am.actions.Response("SHARE_PROVIDERS_RESPONSE", this._shareProvider.socialProviders || []));
```

Comment:
Doesn't social providers return an empty array by default? I see the [return statement](https://github.com/k88hudson/activity-stream/blob/bdcd57107194a4925a59c6848fd4c7aabbed6890/addon/ShareProvider.js#L147) is a map, even if empty you still get `[]` and if it's undefined it will crash either way.
I personally dislike this kind of defensive programming because it can propagate bugs away from the real source and make you chase them in different places in your codebase.